src/plugins/kibana/server/routes/api/ingest/register_simulate.js

```diff
@@ -0,0 +1,25 @@
+const _ = require('lodash');
+const buildRequest = require('../../../lib/ingest_build_request');
+const processResponse = require('../../../lib/ingest_process_response');
+
+module.exports = function registerSimulate(server) {
+  server.route({
+    path: '/api/kibana/ingest/simulate',
+    method: 'POST',
```

Comment:
A Joi schema here would be really good for both validating the request payload and helping other devs understand what the payload should look like. Something like this: https://github.com/elastic/kibana/blob/master/src/plugins/kibana/server/routes/api/ingest/register_post.js#L14